go/tdh2/tdh2/tdh2_test.go

```diff
@@ -563,6 +561,17 @@ func TestCheckEi(t *testing.T) {
 				},
 				err: cmpopts.AnyError,
 			},
+			{
+				name: "negative share index",
+				ctxt: ctxt,
+				share: &DecryptionShare{
+					index: -1,
+					u_i:   ds.u_i,
+					e_i:   ds.e_i,
+					f_i:   ds.f_i,
+				},
+				err: cmpopts.AnyError,
```

Comment:
Not a big deal, but I recommend tightening these error types up to prevent false negatives resulting from some unanticipated error.